model/semaphore.rb

```diff
@@ -7,8 +7,15 @@ class Semaphore < Sequel::Model
 
   def self.incr(strand_id, name)
     DB.transaction do
-      Strand.dataset.where(id: strand_id).update(schedule: Sequel::CURRENT_TIMESTAMP)
-      Semaphore.create_with_id(strand_id: strand_id, name: name)
+      if Strand.where(id: strand_id).update(schedule: Sequel::CURRENT_TIMESTAMP) == 1
+        create(strand_id:, name:)
+      end
+    end
```

Comment:
So this one is probably safe even without the rescue, because it's in a transaction and update takes a lock for the duration of the transaction that blocks a delete. The previous code I read in the context of autocommit, where an UPDATE commits, and then the record disappears.

Reply:
Good point.  I had a spec that tested this, but only by mocking things so we deleted the strand before the creation inside the same transaction.  Since that cannot occur in production, I agree the rescue is unnecessary.  I've pushed an updated commit to remove it the rescue and related spec.